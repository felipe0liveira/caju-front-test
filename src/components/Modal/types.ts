export type ModalStateProps = {
  title?: string;
  content?: JSX.Element | string;
  footerVisible?: boolean;
} & FooterProps;

export interface ContentProps {
  children: JSX.Element;
  title?: string;
  onClose: () => void;
}

export interface FooterProps {
  message?: string;
  onConfirmText?: string;
  onCancelText?: string;
  onConfirmCallback?: () => void;
  onCancelCallback?: () => void;
}
