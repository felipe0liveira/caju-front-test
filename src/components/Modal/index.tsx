import { HiOutlineX } from "react-icons/hi";
import { ContentProps, FooterProps } from "./types";
import { ButtonSmall } from "../Buttons";
import {
  Body,
  Container,
  FooterActions,
  FooterContainer,
  Header,
} from "./styles";
import { BackgroundContainer } from "../SharedStyles";

const Content = (props: ContentProps) => {
  return (
    <BackgroundContainer onClick={props.onClose}>
      <Container onClick={(e: Event) => e.stopPropagation()}>
        <Header>
          <h4>{props.title}</h4>
          <HiOutlineX className="close" onClick={props.onClose} />
        </Header>
        <Body>{props.children}</Body>
      </Container>
    </BackgroundContainer>
  );
};

const Footer = (props: FooterProps) => {
  return (
    <FooterContainer>
      <span>{props.message}</span>

      <FooterActions>
        {props.onCancelText && props.onConfirmCallback && (
          <ButtonSmall
            bgcolor="rgb(255, 145, 154)"
            onClick={props.onCancelCallback}
          >
            {props.onCancelText}
          </ButtonSmall>
        )}
        {props.onConfirmText && props.onConfirmCallback && (
          <ButtonSmall
            bgcolor="rgb(155, 229, 155)"
            onClick={props.onConfirmCallback}
          >
            {props.onConfirmText}
          </ButtonSmall>
        )}
      </FooterActions>
    </FooterContainer>
  );
};

export const Modal = {
  Content,
  Footer,
};

export default Modal;
