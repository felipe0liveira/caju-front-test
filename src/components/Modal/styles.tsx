import styled from "styled-components";

export const Container = styled.div`
  background: rgba(255, 255, 255, 0.95);
  border-radius: 32px;
  box-shadow: 0 0 300px rgba(0, 0, 0, 0.3);

  width: 500px;
  max-width: 80%;

  height: auto;
  min-height: 100px;
  padding: 32px;
`;

export const Header = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;

  h4 {
    margin: 0;
  }

  .close {
    cursor: pointer;
  }
`;

export const Body = styled.div`
  margin-top: 16px;
`;

export const FooterContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;

  width: 100%;
  margin-top: 32px;

  span {
    color: rgba(0, 0, 0, 0.5);
    font-size: 14px;
  }
`;

export const FooterActions = styled.div`
  display: flex;
  align-items: center;
  justify-content: end;
  gap: 16px;
`;
