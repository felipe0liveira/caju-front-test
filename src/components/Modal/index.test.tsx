import Modal from ".";
import { render } from "@testing-library/react";
import { describe, test, expect } from "@jest/globals";

describe("<Modal />", () => {
  test("Should match snapshot", () => {
    const component = render(
      <Modal.Content onClose={() => {}}>
        <>
          <p>A modal content</p>
          <Modal.Footer />
        </>
      </Modal.Content>
    );
    expect(component).toMatchSnapshot();
  });
});
