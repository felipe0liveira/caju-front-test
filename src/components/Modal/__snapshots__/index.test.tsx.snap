// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-kAycRU iFYosF"
      >
        <div
          class="sc-eqUzNf dmHEfC"
        >
          <div
            class="sc-fqkwJk ievLso"
          >
            <h4 />
            <svg
              aria-hidden="true"
              class="close"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 18L18 6M6 6l12 12"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
          <div
            class="sc-dcJtft kWiOqI"
          >
            <p>
              A modal content
            </p>
            <div
              class="sc-iGgVNO iOcrXN"
            >
              <span />
              <div
                class="sc-gsFSjX kkSBQc"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-kAycRU iFYosF"
    >
      <div
        class="sc-eqUzNf dmHEfC"
      >
        <div
          class="sc-fqkwJk ievLso"
        >
          <h4 />
          <svg
            aria-hidden="true"
            class="close"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18L18 6M6 6l12 12"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        <div
          class="sc-dcJtft kWiOqI"
        >
          <p>
            A modal content
          </p>
          <div
            class="sc-iGgVNO iOcrXN"
          >
            <span />
            <div
              class="sc-gsFSjX kkSBQc"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
